src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -865,8 +865,9 @@ _stream_run_hello (mongoc_cluster_t *cluster,
 
    rtt_msec = (bson_get_monotonic_time () - start) / 1000;
 
-   ret_handshake_sd = (mongoc_server_description_t *) bson_malloc0 (
-      sizeof (mongoc_server_description_t));
+   ret_handshake_sd =
+      bson_aligned_alloc0 (BSON_ALIGNOF (mongoc_server_description_t),
+                           sizeof (mongoc_server_description_t));
```

Comment:
This pattern of `aligned_alloc(alignof(T), sizeof(T))` is pretty repetitive. It might be useful to have a `bson_alloc_new(T)` function-like macro.

Reply:
Good call. Added `BSON_ALIGNED_ALLOC` and `BSON_ALIGNED_ALLOC0` convenience macros to avoid type repetition.